File: archive/t/typescript/longest-word.ts
if (process.argv.length < 3 || process.argv[2].length < 1) {
    console.log("Usage: please provide a string")
    process.exit(0)
}

let testString: string = process.argv[2];

//     const separatedWords: string[] = input.split(' ');

//     let longestWord: string = "";
//     let wordLength: number = 0;

//     for (let word of separatedWords) {
//         if (word.length > longestWord.length) {
//             longestWord = word;
//             wordLength = word.length;
//         }
//         else if (word == "") {
//             console.log(error_message);
//             process.exit(1);
//         }
//     }

//     return wordLength;
// }

File: archive/p/perl/longest-word.pl
#!/usr/bin/env perl
use v5.42;
use List::Util qw/max/;

sub usage { say "Usage: please provide a string"; exit }

my $input = "@ARGV";
$input =~ s/^\s+|\s+$//g;

usage unless length $input;
say max map length, split /\s+/, $input;